spec/tasks/coa_upgrade/task_spec.rb

```diff
@@ -0,0 +1,108 @@
+require 'spec_helper'
+require 'tmpdir'
+require 'yaml'
+require_relative '../task_spec_helper'
+
+describe 'COA upgrade task' do
+  let(:yaml_files) { File.join('**', '*.yml') }
+  let(:stderr_path) { File.join(@upgrate_results, 'stderr.log') }
+  let(:stderr_loaded_file) do
+    content = File.read(stderr_path) if File.exist?(stderr_path)
+    content || ''
+  end
+
+  before(:context) do
```

Comment:
I had a bad gut feeling about instancing so many ivars in test cases, what I usually always go long way to avoid because of side effects. OTOH, I don't how to do it differently because of the cp_r. I'll give it a quick shot.